# **1. Import Libraries**

In [103]:
import numpy as np
import polars as pl
import pandas as pd

# **2. Load Datasets**

In [104]:
data1_pl = pl.read_csv(
    'dataset1_completion_rate.csv',
    separator=";",
    skip_rows=2,              
)

data2_pl = pl.read_csv(
    'dataset2_cohort_survival_rate.csv',
    separator=";",
    skip_rows=2,              
)

data3_pl = pl.read_csv(
    'dataset3_gender_parity_index.csv',
    separator=";",
    skip_rows=2,              
)

data4_pl = pl.read_csv(
    'dataset4_participation_rate.csv',
    separator=";",
    skip_rows=2,              
)

# **3. Data Cleaning and Preperation**

## *Dataset 1: Completion Rate* 

### 1. Dimensionality Reduction

In [105]:
data1_pl = data1_pl.drop('Indicator','2000','2001','2024','2025')
data1_pl = data1_pl.filter((pl.col('Geolocation') != 'PHILIPPINES') & (pl.col('Sex') != 'Both Sexes'))

data1_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ str  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 71.6 ┆ 99.8  ┆ 93.5 ┆ 95.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.3 ┆ 99.7  ┆ 94.7 ┆ 93.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.1 ┆ … ┆ 77.6 ┆ 100.0 ┆ 97.3 ┆ 92.5 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 68.9 ┆ … ┆ 69.9 ┆ 100.0 ┆ 98.4 ┆ 86.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ ..   ┆ … ┆ 61.9 ┆ 98.2  ┆ 85.5 ┆ 85.4 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 41.0 ┆ … ┆ 61.1 ┆ 84.1  ┆ 89.7 ┆ 62.5 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 71.5 ┆ … ┆ 77.5 ┆ 97.2  ┆ 85.5 ┆ 73.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 49.7 ┆ … ┆ 67.4 ┆ 93.5  ┆ 83.7 ┆ 60.4 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ ..   ┆ … ┆ 61.2 ┆ 92.9  ┆ 65.8 ┆ 57.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ ..   ┆ … ┆ 51.9 ┆ 86.6  ┆ 63.8 ┆ 46.0 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 2. Imputing Null Values

In [106]:
data1_pl = data1_pl.with_columns(
    pl.when(pl.col('Level of Education') == 'Secondary (Senior High School)')
    .then(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']).str.replace_all("..",'0'))
    .otherwise(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']))
)

data1_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ str  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 71.6 ┆ 99.8  ┆ 93.5 ┆ 95.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.3 ┆ 99.7  ┆ 94.7 ┆ 93.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.1 ┆ … ┆ 77.6 ┆ 100.0 ┆ 97.3 ┆ 92.5 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 68.9 ┆ … ┆ 69.9 ┆ 100.0 ┆ 98.4 ┆ 86.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ 0    ┆ … ┆ 61.9 ┆ 98.2  ┆ 85.5 ┆ 85.4 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 41.0 ┆ … ┆ 61.1 ┆ 84.1  ┆ 89.7 ┆ 62.5 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 71.5 ┆ … ┆ 77.5 ┆ 97.2  ┆ 85.5 ┆ 73.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 49.7 ┆ … ┆ 67.4 ┆ 93.5  ┆ 83.7 ┆ 60.4 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ 0    ┆ … ┆ 61.2 ┆ 92.9  ┆ 65.8 ┆ 57.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ 0    ┆ … ┆ 51.9 ┆ 86.6  ┆ 63.8 ┆ 46.0 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 3. Standardizing Columns

In [107]:
data1_pl = data1_pl.with_columns(
    pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']).cast(pl.Float64)
)

data1_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ f64  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 71.6 ┆ 99.8  ┆ 93.5 ┆ 95.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.3 ┆ 99.7  ┆ 94.7 ┆ 93.4 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.1 ┆ … ┆ 77.6 ┆ 100.0 ┆ 97.3 ┆ 92.5 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 68.9 ┆ … ┆ 69.9 ┆ 100.0 ┆ 98.4 ┆ 86.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ 0.0  ┆ … ┆ 61.9 ┆ 98.2  ┆ 85.5 ┆ 85.4 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 41.0 ┆ … ┆ 61.1 ┆ 84.1  ┆ 89.7 ┆ 62.5 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 71.5 ┆ … ┆ 77.5 ┆ 97.2  ┆ 85.5 ┆ 73.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 49.7 ┆ … ┆ 67.4 ┆ 93.5  ┆ 83.7 ┆ 60.4 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ 0.0  ┆ … ┆ 61.2 ┆ 92.9  ┆ 65.8 ┆ 57.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ 0.0  ┆ … ┆ 51.9 ┆ 86.6  ┆ 63.8 ┆ 46.0 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 4. Simplify Feature Names

In [108]:
# Geolocation

'''
..National Capital Region (NCR) = NCR
..Cordillera Administrative Region (CAR) = CAR
..MIMAROPA Region = Region IV-B
..Bangsamoro Autonomous Region in Muslim Mindanao (BARMM) = BARMM

'''

data1_pl = data1_pl.with_columns(
    pl.col('Geolocation').str.replace('..','')
)

data1_pl = data1_pl.with_columns(
    pl.col("Geolocation").replace({'National Capital Region (NCR)':'NCR',
   'Cordillera Administrative Region (CAR)':'CAR',
   'MIMAROPA Region':'Region IV-B',
   'Bangsamoro Autonomous Region in Muslim Mindanao (BARMM)':'BARMM'})
)


In [109]:
# Level of Education

'''
Elementary = Elementary
Secondary (Junior High School) = Junior High
Secondary (Senior High School) = Senior High

'''

data1_pl = data1_pl.with_columns(
    pl.col("Level of Education").replace({'Secondary (Junior High School)':'Junior High',
   'Secondary (Senior High School)':'Senior High'})
)

data1_pl


shape: (102, 25)
┌─────────────┬────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation ┆ Level of Education ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---         ┆ ---                ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str         ┆ str                ┆ str    ┆ f64  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞═════════════╪════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ NCR         ┆ Elementary         ┆ Female ┆ 87.7 ┆ … ┆ 71.6 ┆ 99.8  ┆ 93.5 ┆ 95.4 │
│ NCR         ┆ Elementary         ┆ Male   ┆ 81.3 ┆ … ┆ 67.3 ┆ 99.7  ┆ 94.7 ┆ 93.4 │
│ NCR         ┆ Junior High        ┆ Female ┆ 82.1 ┆ … ┆ 77.6 ┆ 100.0 ┆ 97.3 ┆ 92.5 │
│ NCR         ┆ Junior High        ┆ Male   ┆ 68.9 ┆ … ┆ 69.9 ┆ 100.0 ┆ 98.4 ┆ 86.2 │
│ NCR         ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 61.9 ┆ 98.2  ┆ 85.5 ┆ 85.4 │
│ …           ┆ …                  ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ BARMM       ┆ Elementary         ┆ Male   ┆ 41.0 ┆ … ┆ 61.1 ┆ 84.1  ┆ 89.7 ┆ 62.5 │
│ BARMM       ┆ Junior High        ┆ Female ┆ 71.5 ┆ … ┆ 77.5 ┆ 97.2  ┆ 85.5 ┆ 73.5 │
│ BARMM       ┆ Junior High        ┆ Male   ┆ 49.7 ┆ … ┆ 67.4 ┆ 93.5  ┆ 83.7 ┆ 60.4 │
│ BARMM       ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 61.2 ┆ 92.9  ┆ 65.8 ┆ 57.1 │
│ BARMM       ┆ Senior High        ┆ Male   ┆ 0.0  ┆ … ┆ 51.9 ┆ 86.6  ┆ 63.8 ┆ 46.0 │
└─────────────┴────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

## *Dataset 2: Cohort Survival Rate*

### 1. Dimensionality Reduction

In [110]:
data2_pl = data2_pl.drop('Indicator','2000','2001','2024','2025')
data2_pl = data2_pl.filter((pl.col('Geolocation') != 'PHILIPPINES') & (pl.col('Sex') != 'Both Sexes'))

data2_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ str  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 72.0 ┆ 100.0 ┆ 93.7 ┆ 95.7 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.8 ┆ 100.0 ┆ 94.7 ┆ 93.8 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.8 ┆ … ┆ 78.5 ┆ 100.0 ┆ 97.4 ┆ 93.6 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 71.1 ┆ … ┆ 70.8 ┆ 100.0 ┆ 98.4 ┆ 87.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ ..   ┆ … ┆ 63.8 ┆ 99.3  ┆ 89.2 ┆ 87.7 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 43.8 ┆ … ┆ 62.4 ┆ 85.5  ┆ 89.8 ┆ 63.7 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 76.7 ┆ … ┆ 78.5 ┆ 99.5  ┆ 86.8 ┆ 75.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 54.0 ┆ … ┆ 68.9 ┆ 95.8  ┆ 84.6 ┆ 62.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ ..   ┆ … ┆ 62.9 ┆ 94.2  ┆ 68.3 ┆ 58.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ ..   ┆ … ┆ 53.8 ┆ 88.2  ┆ 67.0 ┆ 47.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 2. Imputing Null Values

In [111]:
data2_pl = data2_pl.with_columns(
    pl.when(pl.col('Level of Education') == 'Secondary (Senior High School)')
    .then(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']).str.replace_all("..",'0'))
    .otherwise(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']))
)

data2_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ str  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 72.0 ┆ 100.0 ┆ 93.7 ┆ 95.7 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.8 ┆ 100.0 ┆ 94.7 ┆ 93.8 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.8 ┆ … ┆ 78.5 ┆ 100.0 ┆ 97.4 ┆ 93.6 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 71.1 ┆ … ┆ 70.8 ┆ 100.0 ┆ 98.4 ┆ 87.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ 0    ┆ … ┆ 63.8 ┆ 99.3  ┆ 89.2 ┆ 87.7 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 43.8 ┆ … ┆ 62.4 ┆ 85.5  ┆ 89.8 ┆ 63.7 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 76.7 ┆ … ┆ 78.5 ┆ 99.5  ┆ 86.8 ┆ 75.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 54.0 ┆ … ┆ 68.9 ┆ 95.8  ┆ 84.6 ┆ 62.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ 0    ┆ … ┆ 62.9 ┆ 94.2  ┆ 68.3 ┆ 58.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ 0    ┆ … ┆ 53.8 ┆ 88.2  ┆ 67.0 ┆ 47.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 3. Standardizing Columns

In [112]:
data2_pl = data2_pl.with_columns(
    pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017']).cast(pl.Float64)
)

data2_pl

shape: (102, 25)
┌────────────────────┬────────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation        ┆ Level of Education     ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---                ┆ ---                    ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str                ┆ str                    ┆ str    ┆ f64  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞════════════════════╪════════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ ..National Capital ┆ Elementary             ┆ Female ┆ 87.7 ┆ … ┆ 72.0 ┆ 100.0 ┆ 93.7 ┆ 95.7 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Elementary             ┆ Male   ┆ 81.3 ┆ … ┆ 67.8 ┆ 100.0 ┆ 94.7 ┆ 93.8 │
│ Region (NCR…       ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Female ┆ 82.8 ┆ … ┆ 78.5 ┆ 100.0 ┆ 97.4 ┆ 93.6 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Junior High ┆ Male   ┆ 71.1 ┆ … ┆ 70.8 ┆ 100.0 ┆ 98.4 ┆ 87.2 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..National Capital ┆ Secondary (Senior High ┆ Female ┆ 0.0  ┆ … ┆ 63.8 ┆ 99.3  ┆ 89.2 ┆ 87.7 │
│ Region (NCR…       ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ …                  ┆ …                      ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ ..Bangsamoro       ┆ Elementary             ┆ Male   ┆ 43.8 ┆ … ┆ 62.4 ┆ 85.5  ┆ 89.8 ┆ 63.7 │
│ Autonomous Region… ┆                        ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Female ┆ 76.7 ┆ … ┆ 78.5 ┆ 99.5  ┆ 86.8 ┆ 75.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Junior High ┆ Male   ┆ 54.0 ┆ … ┆ 68.9 ┆ 95.8  ┆ 84.6 ┆ 62.5 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Female ┆ 0.0  ┆ … ┆ 62.9 ┆ 94.2  ┆ 68.3 ┆ 58.1 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
│ ..Bangsamoro       ┆ Secondary (Senior High ┆ Male   ┆ 0.0  ┆ … ┆ 53.8 ┆ 88.2  ┆ 67.0 ┆ 47.2 │
│ Autonomous Region… ┆ School)                ┆        ┆      ┆   ┆      ┆       ┆      ┆      │
└────────────────────┴────────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

### 4. Simplifying Feature Names

In [113]:
# Geolocation

'''
..National Capital Region (NCR) = NCR
..Cordillera Administration Region (CAR) = CAR
..MIMAROPA Region = Region IV-B
..Bangsamoro Autonomous Region in Muslim Mindanao (BARMM) = BARMM

'''

data2_pl = data2_pl.with_columns(
    pl.col('Geolocation').str.replace('..','')
)

data2_pl = data2_pl.with_columns(
    pl.col("Geolocation").replace({'National Capital Region (NCR)':'NCR',
   'Cordillera Administration Region (CAR)':'CAR',
   'MIMAROPA Region':'Region IV-B',
   'Bangsamoro Autonomous Region in Muslim Mindanao (BARMM)':'BARMM'})
)

In [114]:
# Level of Education

'''
Elementary = Elementary
Secondary (Junior High School) = Junior High
Secondary (Senior High School) = Senior High

'''

data2_pl = data2_pl.with_columns(
    pl.col("Level of Education").replace({'Secondary (Junior High School)':'Junior High',
   'Secondary (Senior High School)':'Senior High'})
)

data2_pl

shape: (102, 25)
┌─────────────┬────────────────────┬────────┬──────┬───┬──────┬───────┬──────┬──────┐
│ Geolocation ┆ Level of Education ┆ Sex    ┆ 2002 ┆ … ┆ 2020 ┆ 2021  ┆ 2022 ┆ 2023 │
│ ---         ┆ ---                ┆ ---    ┆ ---  ┆   ┆ ---  ┆ ---   ┆ ---  ┆ ---  │
│ str         ┆ str                ┆ str    ┆ f64  ┆   ┆ f64  ┆ f64   ┆ f64  ┆ f64  │
╞═════════════╪════════════════════╪════════╪══════╪═══╪══════╪═══════╪══════╪══════╡
│ NCR         ┆ Elementary         ┆ Female ┆ 87.7 ┆ … ┆ 72.0 ┆ 100.0 ┆ 93.7 ┆ 95.7 │
│ NCR         ┆ Elementary         ┆ Male   ┆ 81.3 ┆ … ┆ 67.8 ┆ 100.0 ┆ 94.7 ┆ 93.8 │
│ NCR         ┆ Junior High        ┆ Female ┆ 82.8 ┆ … ┆ 78.5 ┆ 100.0 ┆ 97.4 ┆ 93.6 │
│ NCR         ┆ Junior High        ┆ Male   ┆ 71.1 ┆ … ┆ 70.8 ┆ 100.0 ┆ 98.4 ┆ 87.2 │
│ NCR         ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 63.8 ┆ 99.3  ┆ 89.2 ┆ 87.7 │
│ …           ┆ …                  ┆ …      ┆ …    ┆ … ┆ …    ┆ …     ┆ …    ┆ …    │
│ BARMM       ┆ Elementary         ┆ Male   ┆ 43.8 ┆ … ┆ 62.4 ┆ 85.5  ┆ 89.8 ┆ 63.7 │
│ BARMM       ┆ Junior High        ┆ Female ┆ 76.7 ┆ … ┆ 78.5 ┆ 99.5  ┆ 86.8 ┆ 75.2 │
│ BARMM       ┆ Junior High        ┆ Male   ┆ 54.0 ┆ … ┆ 68.9 ┆ 95.8  ┆ 84.6 ┆ 62.5 │
│ BARMM       ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 62.9 ┆ 94.2  ┆ 68.3 ┆ 58.1 │
│ BARMM       ┆ Senior High        ┆ Male   ┆ 0.0  ┆ … ┆ 53.8 ┆ 88.2  ┆ 67.0 ┆ 47.2 │
└─────────────┴────────────────────┴────────┴──────┴───┴──────┴───────┴──────┴──────┘

## *Dataset 3: Gender Parity Index*

### 1. Dimensionality Reduction

In [115]:
data3_pl = data3_pl.drop('2000','2001','2024','2025')
data3_pl = data3_pl.filter((pl.col('Geolocation') != 'PHILIPPINES') & (pl.col('Level of education') != 'Ratio of girls to boys in tertiary education'))

data3_pl


shape: (51, 24)
┌──────────────────────────┬─────────────────────────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation              ┆ Level of education      ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                      ┆ ---                     ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                      ┆ str                     ┆ str  ┆ str  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ str  │
╞══════════════════════════╪═════════════════════════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Region (NCR…             ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 1.0  ┆ 1.0  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ ..   ┆ ..   ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.1  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Administrative Re…       ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Administrative Re…       ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ …                        ┆ …                       ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ ..   ┆ ..   ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region…                  ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.3  ┆ 1.2  ┆ … ┆ 1.2  ┆ 1.2  ┆ 1.2  ┆ 1.2  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ ..   ┆ ..   ┆ … ┆ 1.3  ┆ 1.3  ┆ 1.3  ┆ 1.3  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
└──────────────────────────┴─────────────────────────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 2. Imputing Null Values

In [116]:
data3_pl = data3_pl.with_columns(
    pl.when(pl.col('Level of education') == 'Ratio of girls to boys in secondary education (SHS)')
    .then(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015']).str.replace_all("..",'0'))
    .otherwise(pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015']))
)

data3_pl

shape: (51, 24)
┌──────────────────────────┬─────────────────────────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation              ┆ Level of education      ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                      ┆ ---                     ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                      ┆ str                     ┆ str  ┆ str  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ str  │
╞══════════════════════════╪═════════════════════════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Region (NCR…             ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 1.0  ┆ 1.0  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 0    ┆ 0    ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.1  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Administrative Re…       ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Administrative Re…       ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ …                        ┆ …                       ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ 0    ┆ 0    ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region…                  ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.3  ┆ 1.2  ┆ … ┆ 1.2  ┆ 1.2  ┆ 1.2  ┆ 1.2  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 0    ┆ 0    ┆ … ┆ 1.3  ┆ 1.3  ┆ 1.3  ┆ 1.3  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
└──────────────────────────┴─────────────────────────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 3. Standardizing Columns

In [117]:
data3_pl = data3_pl.with_columns(
    pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009','2010','2011','2012','2013','2014','2015','2016','2017','2023']).cast(pl.Float64)
)

data3_pl

shape: (51, 24)
┌──────────────────────────┬─────────────────────────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation              ┆ Level of education      ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                      ┆ ---                     ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                      ┆ str                     ┆ f64  ┆ f64  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞══════════════════════════╪═════════════════════════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Region (NCR…             ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 1.0  ┆ 1.0  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..National Capital       ┆ Ratio of girls to boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.1  ┆ 1.0  │
│ Region (NCR…             ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ Administrative Re…       ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Cordillera             ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Administrative Re…       ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ …                        ┆ …                       ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Region XIII            ┆ Ratio of girls to boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.0  ┆ 1.0  │
│                          ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region…                  ┆ in prim…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 1.3  ┆ 1.2  ┆ … ┆ 1.2  ┆ 1.2  ┆ 1.2  ┆ 1.2  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
│ ..Bangsamoro Autonomous  ┆ Ratio of girls to boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.3  ┆ 1.3  ┆ 1.3  ┆ 1.3  │
│ Region…                  ┆ in seco…                ┆      ┆      ┆   ┆      ┆      ┆      ┆      │
└──────────────────────────┴─────────────────────────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 4. Simplifying Feature Names

In [118]:
# Geolocation

'''
..National Capital Region (NCR) = NCR
..Cordillera Administration Region (CAR) = CAR
..MIMAROPA Region = Region IV-B
..Bangsamoro Autonomous Region in Muslim Mindanao (BARMM) = BARMM

'''

data3_pl = data3_pl.with_columns(
    pl.col('Geolocation').str.replace('..','')
)

data3_pl = data3_pl.with_columns(
    pl.col("Geolocation").replace({'National Capital Region (NCR)':'NCR',
   'Cordillera Administrative Region (CAR)':'CAR',
   'MIMAROPA Region':'Region IV-B',
   'Bangsamoro Autonomous Region in Muslim Mindanao (BARMM)':'BARMM'})
)

In [119]:
# Level of education

'''
Ratio of girls to boys in primary education = Elementary
Ratio of girls to boys in secondary education (JHS) = Junior High
Ratio of girls to boys in secondary education (SHS) = Senior High

'''
data3_pl = data3_pl.rename({'Level of education':'Level of Education'})

data3_pl = data3_pl.with_columns(
    pl.col("Level of Education").replace({'Ratio of girls to boys in primary education':'Elementary',
   'Ratio of girls to boys in secondary education (JHS)':'Junior High',
   'Ratio of girls to boys in secondary education (SHS)':'Senior High'})
)

data3_pl


shape: (51, 24)
┌─────────────┬────────────────────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation ┆ Level of Education ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---         ┆ ---                ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str         ┆ str                ┆ f64  ┆ f64  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞═════════════╪════════════════════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ NCR         ┆ Elementary         ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ NCR         ┆ Junior High        ┆ 1.0  ┆ 1.0  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ NCR         ┆ Senior High        ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.1  ┆ 1.0  │
│ CAR         ┆ Elementary         ┆ 0.9  ┆ 0.9  ┆ … ┆ 0.9  ┆ 0.9  ┆ 0.9  ┆ 0.9  │
│ CAR         ┆ Junior High        ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ …           ┆ …                  ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ Region XIII ┆ Junior High        ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ Region XIII ┆ Senior High        ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.1  ┆ 1.1  ┆ 1.0  ┆ 1.0  │
│ BARMM       ┆ Elementary         ┆ 1.1  ┆ 1.1  ┆ … ┆ 1.0  ┆ 1.0  ┆ 1.0  ┆ 1.0  │
│ BARMM       ┆ Junior High        ┆ 1.3  ┆ 1.2  ┆ … ┆ 1.2  ┆ 1.2  ┆ 1.2  ┆ 1.2  │
│ BARMM       ┆ Senior High        ┆ 0.0  ┆ 0.0  ┆ … ┆ 1.3  ┆ 1.3  ┆ 1.3  ┆ 1.3  │
└─────────────┴────────────────────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

## *Dataset 4: Participation Rate*

### 1. Dimensionality Reduction

In [120]:
data4_pl = data4_pl.drop('Indicator','2000','2001','2024','2025')
data4_pl = data4_pl.filter((pl.col('Geolocation') != 'PHILIPPINES') & (pl.col('Sex') != 'Both Sexes'))

data4_pl

shape: (34, 24)
┌─────────────────────────────────┬───────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation                     ┆ Sex   ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                             ┆ ---   ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                             ┆ str   ┆ str  ┆ str  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞═════════════════════════════════╪═══════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital Region (NCR… ┆ Girls ┆ ..   ┆ ..   ┆ … ┆ 59.2 ┆ 67.9 ┆ 69.8 ┆ 73.4 │
│ ..National Capital Region (NCR… ┆ Boys  ┆ ..   ┆ ..   ┆ … ┆ 54.3 ┆ 62.4 ┆ 63.6 ┆ 67.5 │
│ ..Cordillera Administrative Re… ┆ Girls ┆ ..   ┆ ..   ┆ … ┆ 86.4 ┆ 87.9 ┆ 87.1 ┆ 86.1 │
│ ..Cordillera Administrative Re… ┆ Boys  ┆ ..   ┆ ..   ┆ … ┆ 82.5 ┆ 84.7 ┆ 84.2 ┆ 83.3 │
│ ..Region I                      ┆ Girls ┆ ..   ┆ ..   ┆ … ┆ 81.7 ┆ 81.4 ┆ 80.9 ┆ 80.3 │
│ …                               ┆ …     ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XII                    ┆ Boys  ┆ ..   ┆ ..   ┆ … ┆ 65.2 ┆ 71.4 ┆ 72.9 ┆ 73.7 │
│ ..Region XIII                   ┆ Girls ┆ ..   ┆ ..   ┆ … ┆ 79.1 ┆ 81.8 ┆ 82.8 ┆ 82.4 │
│ ..Region XIII                   ┆ Boys  ┆ ..   ┆ ..   ┆ … ┆ 76.1 ┆ 78.9 ┆ 80.5 ┆ 80.2 │
│ ..Bangsamoro Autonomous Region… ┆ Girls ┆ ..   ┆ ..   ┆ … ┆ 49.4 ┆ 50.0 ┆ 57.1 ┆ 57.3 │
│ ..Bangsamoro Autonomous Region… ┆ Boys  ┆ ..   ┆ ..   ┆ … ┆ 44.9 ┆ 45.0 ┆ 52.3 ┆ 52.2 │
└─────────────────────────────────┴───────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 2. Imputing Null Values

In [121]:
data4_pl = data4_pl.with_columns(
 
    pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009']).str.replace_all("..",'0')
)

data4_pl

shape: (34, 24)
┌─────────────────────────────────┬───────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation                     ┆ Sex   ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                             ┆ ---   ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                             ┆ str   ┆ str  ┆ str  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞═════════════════════════════════╪═══════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital Region (NCR… ┆ Girls ┆ 0    ┆ 0    ┆ … ┆ 59.2 ┆ 67.9 ┆ 69.8 ┆ 73.4 │
│ ..National Capital Region (NCR… ┆ Boys  ┆ 0    ┆ 0    ┆ … ┆ 54.3 ┆ 62.4 ┆ 63.6 ┆ 67.5 │
│ ..Cordillera Administrative Re… ┆ Girls ┆ 0    ┆ 0    ┆ … ┆ 86.4 ┆ 87.9 ┆ 87.1 ┆ 86.1 │
│ ..Cordillera Administrative Re… ┆ Boys  ┆ 0    ┆ 0    ┆ … ┆ 82.5 ┆ 84.7 ┆ 84.2 ┆ 83.3 │
│ ..Region I                      ┆ Girls ┆ 0    ┆ 0    ┆ … ┆ 81.7 ┆ 81.4 ┆ 80.9 ┆ 80.3 │
│ …                               ┆ …     ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XII                    ┆ Boys  ┆ 0    ┆ 0    ┆ … ┆ 65.2 ┆ 71.4 ┆ 72.9 ┆ 73.7 │
│ ..Region XIII                   ┆ Girls ┆ 0    ┆ 0    ┆ … ┆ 79.1 ┆ 81.8 ┆ 82.8 ┆ 82.4 │
│ ..Region XIII                   ┆ Boys  ┆ 0    ┆ 0    ┆ … ┆ 76.1 ┆ 78.9 ┆ 80.5 ┆ 80.2 │
│ ..Bangsamoro Autonomous Region… ┆ Girls ┆ 0    ┆ 0    ┆ … ┆ 49.4 ┆ 50.0 ┆ 57.1 ┆ 57.3 │
│ ..Bangsamoro Autonomous Region… ┆ Boys  ┆ 0    ┆ 0    ┆ … ┆ 44.9 ┆ 45.0 ┆ 52.3 ┆ 52.2 │
└─────────────────────────────────┴───────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 3. Standardizing Columns

In [122]:
data4_pl = data4_pl.with_columns(
    pl.col(['2002','2003','2004','2005','2006','2007',
    '2008','2009']).cast(pl.Float64)
)

data4_pl

shape: (34, 24)
┌─────────────────────────────────┬───────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation                     ┆ Sex   ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---                             ┆ ---   ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str                             ┆ str   ┆ f64  ┆ f64  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞═════════════════════════════════╪═══════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ ..National Capital Region (NCR… ┆ Girls ┆ 0.0  ┆ 0.0  ┆ … ┆ 59.2 ┆ 67.9 ┆ 69.8 ┆ 73.4 │
│ ..National Capital Region (NCR… ┆ Boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 54.3 ┆ 62.4 ┆ 63.6 ┆ 67.5 │
│ ..Cordillera Administrative Re… ┆ Girls ┆ 0.0  ┆ 0.0  ┆ … ┆ 86.4 ┆ 87.9 ┆ 87.1 ┆ 86.1 │
│ ..Cordillera Administrative Re… ┆ Boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 82.5 ┆ 84.7 ┆ 84.2 ┆ 83.3 │
│ ..Region I                      ┆ Girls ┆ 0.0  ┆ 0.0  ┆ … ┆ 81.7 ┆ 81.4 ┆ 80.9 ┆ 80.3 │
│ …                               ┆ …     ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ ..Region XII                    ┆ Boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 65.2 ┆ 71.4 ┆ 72.9 ┆ 73.7 │
│ ..Region XIII                   ┆ Girls ┆ 0.0  ┆ 0.0  ┆ … ┆ 79.1 ┆ 81.8 ┆ 82.8 ┆ 82.4 │
│ ..Region XIII                   ┆ Boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 76.1 ┆ 78.9 ┆ 80.5 ┆ 80.2 │
│ ..Bangsamoro Autonomous Region… ┆ Girls ┆ 0.0  ┆ 0.0  ┆ … ┆ 49.4 ┆ 50.0 ┆ 57.1 ┆ 57.3 │
│ ..Bangsamoro Autonomous Region… ┆ Boys  ┆ 0.0  ┆ 0.0  ┆ … ┆ 44.9 ┆ 45.0 ┆ 52.3 ┆ 52.2 │
└─────────────────────────────────┴───────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

### 4. Simplifying Feature Names

In [123]:
# Geolocation


'''
..National Capital Region (NCR) = NCR
..Cordillera Administration Region (CAR) = CAR
..MIMAROPA Region = Region IV-B
..Bangsamoro Autonomous Region in Muslim Mindanao (BARMM) = BARMM

'''

data4_pl = data4_pl.with_columns(
    pl.col('Geolocation').str.replace('..','')
)

data4_pl = data4_pl.with_columns(
    pl.col("Geolocation").replace({'National Capital Region (NCR)':'NCR',
   'Cordillera Administrative Region (CAR)':'CAR',
   'MIMAROPA Region':'Region IV-B',
   'Bangsamoro Autonomous Region in Muslim Mindanao (BARMM)':'BARMM'})
)

In [124]:
# Sex

'''
Girls = Female
Boys = Male

'''

data4_pl = data4_pl.with_columns(
    pl.col("Sex").replace({'Girls':'Female',
   'Boys':'Male'})
)

data4_pl

shape: (34, 24)
┌─────────────┬────────┬──────┬──────┬───┬──────┬──────┬──────┬──────┐
│ Geolocation ┆ Sex    ┆ 2002 ┆ 2003 ┆ … ┆ 2020 ┆ 2021 ┆ 2022 ┆ 2023 │
│ ---         ┆ ---    ┆ ---  ┆ ---  ┆   ┆ ---  ┆ ---  ┆ ---  ┆ ---  │
│ str         ┆ str    ┆ f64  ┆ f64  ┆   ┆ f64  ┆ f64  ┆ f64  ┆ f64  │
╞═════════════╪════════╪══════╪══════╪═══╪══════╪══════╪══════╪══════╡
│ NCR         ┆ Female ┆ 0.0  ┆ 0.0  ┆ … ┆ 59.2 ┆ 67.9 ┆ 69.8 ┆ 73.4 │
│ NCR         ┆ Male   ┆ 0.0  ┆ 0.0  ┆ … ┆ 54.3 ┆ 62.4 ┆ 63.6 ┆ 67.5 │
│ CAR         ┆ Female ┆ 0.0  ┆ 0.0  ┆ … ┆ 86.4 ┆ 87.9 ┆ 87.1 ┆ 86.1 │
│ CAR         ┆ Male   ┆ 0.0  ┆ 0.0  ┆ … ┆ 82.5 ┆ 84.7 ┆ 84.2 ┆ 83.3 │
│ Region I    ┆ Female ┆ 0.0  ┆ 0.0  ┆ … ┆ 81.7 ┆ 81.4 ┆ 80.9 ┆ 80.3 │
│ …           ┆ …      ┆ …    ┆ …    ┆ … ┆ …    ┆ …    ┆ …    ┆ …    │
│ Region XII  ┆ Male   ┆ 0.0  ┆ 0.0  ┆ … ┆ 65.2 ┆ 71.4 ┆ 72.9 ┆ 73.7 │
│ Region XIII ┆ Female ┆ 0.0  ┆ 0.0  ┆ … ┆ 79.1 ┆ 81.8 ┆ 82.8 ┆ 82.4 │
│ Region XIII ┆ Male   ┆ 0.0  ┆ 0.0  ┆ … ┆ 76.1 ┆ 78.9 ┆ 80.5 ┆ 80.2 │
│ BARMM       ┆ Female ┆ 0.0  ┆ 0.0  ┆ … ┆ 49.4 ┆ 50.0 ┆ 57.1 ┆ 57.3 │
│ BARMM       ┆ Male   ┆ 0.0  ┆ 0.0  ┆ … ┆ 44.9 ┆ 45.0 ┆ 52.3 ┆ 52.2 │
└─────────────┴────────┴──────┴──────┴───┴──────┴──────┴──────┴──────┘

## *Final Datasets: **Philippines Primary and Secondary Education Statistics***

### 1. Dataset Merging (1 and 2)

In [125]:
data1_pl = data1_pl.with_columns(
    pl.lit('Completion Rate').alias('Metric')
)

data1_pl

shape: (102, 26)
┌─────────────┬────────────────────┬────────┬──────┬───┬───────┬──────┬──────┬─────────────────┐
│ Geolocation ┆ Level of Education ┆ Sex    ┆ 2002 ┆ … ┆ 2021  ┆ 2022 ┆ 2023 ┆ Metric          │
│ ---         ┆ ---                ┆ ---    ┆ ---  ┆   ┆ ---   ┆ ---  ┆ ---  ┆ ---             │
│ str         ┆ str                ┆ str    ┆ f64  ┆   ┆ f64   ┆ f64  ┆ f64  ┆ str             │
╞═════════════╪════════════════════╪════════╪══════╪═══╪═══════╪══════╪══════╪═════════════════╡
│ NCR         ┆ Elementary         ┆ Female ┆ 87.7 ┆ … ┆ 99.8  ┆ 93.5 ┆ 95.4 ┆ Completion Rate │
│ NCR         ┆ Elementary         ┆ Male   ┆ 81.3 ┆ … ┆ 99.7  ┆ 94.7 ┆ 93.4 ┆ Completion Rate │
│ NCR         ┆ Junior High        ┆ Female ┆ 82.1 ┆ … ┆ 100.0 ┆ 97.3 ┆ 92.5 ┆ Completion Rate │
│ NCR         ┆ Junior High        ┆ Male   ┆ 68.9 ┆ … ┆ 100.0 ┆ 98.4 ┆ 86.2 ┆ Completion Rate │
│ NCR         ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 98.2  ┆ 85.5 ┆ 85.4 ┆ Completion Rate │
│ …           ┆ …                  ┆ …      ┆ …    ┆ … ┆ …     ┆ …    ┆ …    ┆ …               │
│ BARMM       ┆ Elementary         ┆ Male   ┆ 41.0 ┆ … ┆ 84.1  ┆ 89.7 ┆ 62.5 ┆ Completion Rate │
│ BARMM       ┆ Junior High        ┆ Female ┆ 71.5 ┆ … ┆ 97.2  ┆ 85.5 ┆ 73.5 ┆ Completion Rate │
│ BARMM       ┆ Junior High        ┆ Male   ┆ 49.7 ┆ … ┆ 93.5  ┆ 83.7 ┆ 60.4 ┆ Completion Rate │
│ BARMM       ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 92.9  ┆ 65.8 ┆ 57.1 ┆ Completion Rate │
│ BARMM       ┆ Senior High        ┆ Male   ┆ 0.0  ┆ … ┆ 86.6  ┆ 63.8 ┆ 46.0 ┆ Completion Rate │
└─────────────┴────────────────────┴────────┴──────┴───┴───────┴──────┴──────┴─────────────────┘

In [126]:
data2_pl = data2_pl.with_columns(
    pl.lit('Cohort Survival Rate').alias('Metric')
)

data2_pl

shape: (102, 26)
┌─────────────┬────────────────────┬────────┬──────┬───┬───────┬──────┬──────┬─────────────────┐
│ Geolocation ┆ Level of Education ┆ Sex    ┆ 2002 ┆ … ┆ 2021  ┆ 2022 ┆ 2023 ┆ Metric          │
│ ---         ┆ ---                ┆ ---    ┆ ---  ┆   ┆ ---   ┆ ---  ┆ ---  ┆ ---             │
│ str         ┆ str                ┆ str    ┆ f64  ┆   ┆ f64   ┆ f64  ┆ f64  ┆ str             │
╞═════════════╪════════════════════╪════════╪══════╪═══╪═══════╪══════╪══════╪═════════════════╡
│ NCR         ┆ Elementary         ┆ Female ┆ 87.7 ┆ … ┆ 100.0 ┆ 93.7 ┆ 95.7 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ NCR         ┆ Elementary         ┆ Male   ┆ 81.3 ┆ … ┆ 100.0 ┆ 94.7 ┆ 93.8 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ NCR         ┆ Junior High        ┆ Female ┆ 82.8 ┆ … ┆ 100.0 ┆ 97.4 ┆ 93.6 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ NCR         ┆ Junior High        ┆ Male   ┆ 71.1 ┆ … ┆ 100.0 ┆ 98.4 ┆ 87.2 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ NCR         ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 99.3  ┆ 89.2 ┆ 87.7 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ …           ┆ …                  ┆ …      ┆ …    ┆ … ┆ …     ┆ …    ┆ …    ┆ …               │
│ BARMM       ┆ Elementary         ┆ Male   ┆ 43.8 ┆ … ┆ 85.5  ┆ 89.8 ┆ 63.7 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Junior High        ┆ Female ┆ 76.7 ┆ … ┆ 99.5  ┆ 86.8 ┆ 75.2 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Junior High        ┆ Male   ┆ 54.0 ┆ … ┆ 95.8  ┆ 84.6 ┆ 62.5 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 94.2  ┆ 68.3 ┆ 58.1 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Senior High        ┆ Male   ┆ 0.0  ┆ … ┆ 88.2  ┆ 67.0 ┆ 47.2 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
└─────────────┴────────────────────┴────────┴──────┴───┴───────┴──────┴──────┴─────────────────┘

In [127]:
combined1_pl = pl.concat([data1_pl, data2_pl], how="vertical")

combined1_pl

shape: (204, 26)
┌─────────────┬────────────────────┬────────┬──────┬───┬───────┬──────┬──────┬─────────────────┐
│ Geolocation ┆ Level of Education ┆ Sex    ┆ 2002 ┆ … ┆ 2021  ┆ 2022 ┆ 2023 ┆ Metric          │
│ ---         ┆ ---                ┆ ---    ┆ ---  ┆   ┆ ---   ┆ ---  ┆ ---  ┆ ---             │
│ str         ┆ str                ┆ str    ┆ f64  ┆   ┆ f64   ┆ f64  ┆ f64  ┆ str             │
╞═════════════╪════════════════════╪════════╪══════╪═══╪═══════╪══════╪══════╪═════════════════╡
│ NCR         ┆ Elementary         ┆ Female ┆ 87.7 ┆ … ┆ 99.8  ┆ 93.5 ┆ 95.4 ┆ Completion Rate │
│ NCR         ┆ Elementary         ┆ Male   ┆ 81.3 ┆ … ┆ 99.7  ┆ 94.7 ┆ 93.4 ┆ Completion Rate │
│ NCR         ┆ Junior High        ┆ Female ┆ 82.1 ┆ … ┆ 100.0 ┆ 97.3 ┆ 92.5 ┆ Completion Rate │
│ NCR         ┆ Junior High        ┆ Male   ┆ 68.9 ┆ … ┆ 100.0 ┆ 98.4 ┆ 86.2 ┆ Completion Rate │
│ NCR         ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 98.2  ┆ 85.5 ┆ 85.4 ┆ Completion Rate │
│ …           ┆ …                  ┆ …      ┆ …    ┆ … ┆ …     ┆ …    ┆ …    ┆ …               │
│ BARMM       ┆ Elementary         ┆ Male   ┆ 43.8 ┆ … ┆ 85.5  ┆ 89.8 ┆ 63.7 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Junior High        ┆ Female ┆ 76.7 ┆ … ┆ 99.5  ┆ 86.8 ┆ 75.2 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Junior High        ┆ Male   ┆ 54.0 ┆ … ┆ 95.8  ┆ 84.6 ┆ 62.5 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Senior High        ┆ Female ┆ 0.0  ┆ … ┆ 94.2  ┆ 68.3 ┆ 58.1 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
│ BARMM       ┆ Senior High        ┆ Male   ┆ 0.0  ┆ … ┆ 88.2  ┆ 67.0 ┆ 47.2 ┆ Cohort Survival │
│             ┆                    ┆        ┆      ┆   ┆       ┆      ┆      ┆ Rate            │
└─────────────┴────────────────────┴────────┴──────┴───┴───────┴──────┴──────┴─────────────────┘

### 3. Saving Cleaned Dataset

In [128]:
combined1_pl.write_csv("Cleaned_Completion and Cohort Survival Rate.csv")
data3_pl.write_csv("Cleaned_Gender Parity Index.csv")
data4_pl.write_csv("Cleaned_Participation Rate.csv")

# **4. Exploratory Data Analysis**

#### SDG 4 Metrics

- **Completion Rate**: Percentage of Students that finished a Level of Education
- **Cohort Survival Rate**: Percentage of Initial Students from 1st Year that Survived (Did Not Dropout)
- **Gender Parity Index**: Ratio of Girls to Boys on Enrolment Rates (0.9 and Below = Favors Males, 1.1 and Above = Favors Females)
- **Participation Rate**: Percentage of Students that have access to education

#### Available Filters

- **Geolocation**: Specific Regions of Philippines (ex. Region I, Region II, CAR, NCR, etc.)
- **Sex**: Male or Female
- **Level of Education**: Elementary, Junior High, or Senior High

##### Additional Note/s:

- 2017 and Earlier values are set to 0 for Senior High Level. There is no Senior High yet prior to that year.

